src/audio/module_adapter/module/generic.c

```diff
@@ -599,13 +599,23 @@ void mod_free_all(struct processing_module *mod)
 		list_item_del(&container->list);
```

Comment:
Actually also this list_item_del()-call could be removed now that &res->res_list is also reinitialized at the end of the function when mod_resource_init() is called.